Mathlib/Topology/Connected/PathConnected.lean

```diff
@@ -1215,4 +1227,71 @@ theorem IsOpen.isConnected_iff_isPathConnected {U : Set X} (U_op : IsOpen U) :
+instance : LocallyConnectedSpace X := by
+  refine ⟨forall_imp (fun x h ↦ ⟨fun s ↦ ?_⟩) open_pathConnected_basis⟩
+  refine ⟨fun hs ↦ ?_, fun ⟨u, ⟨hu, hxu, _⟩, hus⟩ ↦ mem_nhds_iff.mpr ⟨u, hus, hu, hxu⟩⟩
+  let ⟨u, ⟨hu, hxu, hu'⟩, hus⟩ := (h.mem_iff' s).mp hs
+  exact ⟨u, ⟨hu, hxu, hu'.isConnected⟩, hus⟩
+
+/-- A space is locally path-connected iff all path components of open subsets are open. -/
+lemma locPathConnectedSpace_iff_isOpen_pathComponentIn {X : Type*} [TopologicalSpace X] :
+    LocPathConnectedSpace X ↔ ∀ (x : X) (u : Set X), IsOpen u → IsOpen (pathComponentIn x u) :=
+  ⟨fun _ _ _ hu ↦ hu.pathComponentIn _, fun h ↦ ⟨fun x ↦ ⟨fun s ↦ by
+    refine ⟨fun hs ↦ ?_, fun ⟨_, ht⟩ ↦ Filter.mem_of_superset ht.1.1 ht.2⟩
+    let ⟨u, hu⟩ := mem_nhds_iff.mp hs
+    exact ⟨pathComponentIn x u, ⟨(h x u hu.2.1).mem_nhds (mem_pathComponentIn_self hu.2.2),
+      isPathConnected_pathComponentIn hu.2.2⟩, pathComponentIn_subset.trans hu.1⟩⟩⟩⟩
+
+/-- A space is locally path-connected iff all path components of open subsets are neighbourhoods. -/
+lemma locPathConnectedSpace_iff_pathComponentIn_mem_nhds {X : Type*} [TopologicalSpace X] :
+    LocPathConnectedSpace X ↔
+    ∀ x : X, ∀ u : Set X, IsOpen u → x ∈ u → pathComponentIn x u ∈ nhds x := by
+  rw [locPathConnectedSpace_iff_isOpen_pathComponentIn]
+  simp_rw [forall_comm (β := Set X), ← imp_forall_iff]
+  refine forall_congr' fun u ↦ imp_congr_right fun _ ↦ ?_
+  exact ⟨fun h x hxu ↦ (h x).mem_nhds (mem_pathComponentIn_self hxu),
+    fun h x ↦ isOpen_iff_mem_nhds.mpr fun y hy ↦
+      pathComponentIn_congr hy ▸ h y <| pathComponentIn_subset hy⟩
+
+/-- Any topology coinduced by a locally path-connected topology is locally path-connected. -/
+lemma LocPathConnectedSpace.coinduced {Y : Type*} (f : X → Y) :
+    @LocPathConnectedSpace Y (.coinduced f ‹_›) := by
+  let _ := TopologicalSpace.coinduced f ‹_›; have hf : Continuous f := continuous_coinduced_rng
+  refine locPathConnectedSpace_iff_isOpen_pathComponentIn.mpr fun y u hu ↦
+    isOpen_coinduced.mpr <| isOpen_iff_mem_nhds.mpr fun x hx ↦ ?_
+  have hx' := preimage_mono pathComponentIn_subset hx
+  refine mem_nhds_iff.mpr ⟨pathComponentIn x (f ⁻¹' u), ?_,
+    (hu.preimage hf).pathComponentIn _, mem_pathComponentIn_self hx'⟩
+  rw [← image_subset_iff, ← pathComponentIn_congr hx]
+  exact ((isPathConnected_pathComponentIn hx').image hf).subset_pathComponentIn
+    ⟨x, mem_pathComponentIn_self hx', rfl⟩ <|
+    (image_mono pathComponentIn_subset).trans <| u.image_preimage_subset f
+
+/-- Quotients of locally path-connected spaces are locally path-connected. -/
+lemma Topology.IsQuotientMap.locPathConnectedSpace {f : X → Y} (h : IsQuotientMap f) :
+    LocPathConnectedSpace Y :=
+  h.2 ▸ LocPathConnectedSpace.coinduced f
+
+/-- Quotients of locally path-connected spaces are locally path-connected. -/
+instance Quot.locPathConnectedSpace {r : X → X → Prop} : LocPathConnectedSpace (Quot r) :=
```

Comment:
Why do you have both a version for `Quot` and `Quotient`? I'm really asking because I don't know.

Reply:
Honestly, I'm not sure either - I just saw it being done like this for compact spaces, with both [Quot.compactSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Compactness/Compact.html#Quot.compactSpace) and [Quotient.compactSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Compactness/Compact.html#Quotient.compactSpace) existing. My understanding is that while `Quot` and `Quotient` are mathematically the same thing, both come up in lean in slightly different contexts, and if you just register the instance for one type it won't be able to infer it for the other. I'm actually using neither instance though, the main theorem I need in further PRs is `LocPathConnectedSpace.coinduced`.